homework/81984/src/parser/visitors.hpp

```diff
@@ -0,0 +1,40 @@
+#ifndef RUBY_IPL_PARSER_VISITORS_HPP_
+#define RUBY_IPL_PARSER_VISITORS_HPP_
+
+#include "nodes.hpp"
+#include "../html/syntax_highlighter.hpp"
+
+struct Program;
+struct Arguments;
+struct FunctionChain;
+struct Block;
+struct Call;
+struct Variable;
+struct Function;
+struct Condition;
+struct Conditions;
+struct Conditional;
+
+struct DisplayVisitor
+{
+  SyntaxHighlighter highlighter;
+  Index tabulation;
+
+  DisplayVisitor();
+  ~DisplayVisitor() = default;
+
+  String Visit(Program* node);
+  String Visit(Arguments* node);
```

Comment:
It might be more efficient to keep the String as a member and append to it, instead of creating a new string on every Visit down the tree.